executor/aggfuncs/func_json_objectagg.go

```diff
@@ -0,0 +1,113 @@
+package aggfuncs
+
+import (
+	"fmt"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/types/json"
+	"github.com/pingcap/tidb/util/chunk"
+	"reflect"
+)
+
+type baseJSONObjectAgg struct {
+	baseAggFunc
+}
+
+type partialResult4JsonObjectAgg struct {
+	entries map[string]interface{}
+}
+
+func (e *baseJSONObjectAgg) AllocPartialResult() PartialResult {
+	p := partialResult4JsonObjectAgg{}
+	p.entries = make(map[string]interface{})
+	return PartialResult(&p)
+}
+
+func (e *baseJSONObjectAgg) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4JsonObjectAgg)(pr)
+	p.entries = make(map[string]interface{})
+}
+
+func (e *baseJSONObjectAgg) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4JsonObjectAgg)(pr)
+	if len(p.entries) == 0 {
+		chk.AppendNull(e.ordinal)
+		return nil
+	}
+
+	// appendBinary does not support some type such as uint8、types.time，so convert is needed here
+	for key, val := range p.entries {
+		switch x := val.(type) {
+		case *types.MyDecimal:
+			float64Val, err := x.ToFloat64()
+			if err != nil {
+				return errors.Trace(err)
+			}
+			p.entries[key] = float64Val
+		case []uint8, types.Time, types.Duration:
+			strVal, err := types.ToString(x)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			p.entries[key] = strVal
+		default:
+			p.entries[key] = val
+		}
+	}
+
+	chk.AppendJSON(e.ordinal, json.CreateBinary(p.entries))
+	return nil
+}
+
+func (e *baseJSONObjectAgg) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
+	p := (*partialResult4JsonObjectAgg)(pr)
+	for _, row := range rowsInGroup {
+		key, err := e.args[0].Eval(row)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		value, err := e.args[1].Eval(row)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		if key.IsNull() {
+			err = errors.New("JSON documents may not contain NULL member names")
```

Comment:
please create an `Error` with the correct error number in parser repo, then use that error here. In MySQL, the error number and error message is:

```sql
MySQL(root@127.0.0.1:test) > select json_objectagg(a, b) from t;
ERROR 3158 (22032): JSON documents may not contain NULL member names.
```